CMakeLists.txt

```diff
@@ -7,12 +7,21 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
     if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.2)
         message(FATAL_ERROR "Your GCC version ${CMAKE_CXX_COMPILER_VERSION} is too old.")
     endif()
-elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+    if (APPLE)
+        # https://software.intel.com/en-us/forums/intel-threading-building-blocks/topic/749200
+        message(WARNING "We had to drop support for GCC on OS X because it caused segfaults when used with tbb. You can continue, but don't hold us responsible for any segmentation faults.")
+        execute_process(COMMAND read -p "Press [enter] to continue anyway")
+    endif()
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.0)
```

Comment:
While we’re at it, we should also update the minimum required Clang version. `4.0` does not support all C++17 feature we currently use (e.g. “Template argument deduction for class templates”). It should be `5.0`.

Reply:
done